### PR TITLE
Update FIPS 140-3 self tests (part 2)

### PR DESCRIPTION
**Pull request by slontis**
Cleanup + remove a few tests that are not required.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Thanks for adding extended tests

**Comment by paulidale** [comment]
It's almost tempting to make "severity: fips change" also enable "extended tests".

**Review by t8m — APPROVED** [review]

**Comment by slontis** [comment]
rebased

File: providers/fips/self_test_data.inc [hunk 1]
```diff
@@ -7,6 +7,15 @@
  * https://www.openssl.org/source/license.html
  */
 
+/*
+ * This file contains self test data required by FIPS 140-3 IG
+ * 10.3.A Cryptographic Algorithm Self test Requirements
+ *
+ * Note that in the 'General CAST requirements': Note33 Allows individual
```
Comment (paulidale) [inline providers/fips/self_test_data.inc:14]: space before 33?

File: providers/fips/self_test_data.inc [hunk 2]
```diff
@@ -629,6 +588,11 @@ static const ST_KAT_PARAM tls13_kdf_client_early_secret_params[] = {
     ST_KAT_PARAM_END()
 };
 
+/*
+ * NOTES:
+ * According to FIPS 140-3 10.3.A Note18: SSH KDF is not required, since it is
```
Comment (paulidale) [inline providers/fips/self_test_data.inc:593]: space before 18?

**Review by paulidale — APPROVED** [review]
With or without the additional spaces but prefer with.

File: providers/fips/self_test_data.inc (hunk 2 above)
Comment (slontis) [reply providers/fips/self_test_data.inc:593]: No that is how it is in the doc

**Review by slontis — COMMENTED** [review]
breaks cross tests again..
To take SHA1 out the other branches need the test_fipsinstall.t change..

File: providers/fips/self_test_data.inc (hunk 1 above)
Comment (slontis) [reply providers/fips/self_test_data.inc:14]: Nup

**Comment by slontis** [comment]
Hmm Are reviewers ok with this going to all branches?

**Comment by paulidale** [comment]
I don't see a compelling requirement to back port this.
For 3.0, the requirements are quite different and 3.1 is missing a lot of the other recent activity.

I won't object to the changes being back ported.

**Comment by slontis** [comment]
@paulidale Do you see the failing tests? That is the reason for the backport

**Comment by paulidale** [comment]
Yeah, backport then.  I'd forgotten this wasn't evp_test where we could condition out failing stuff.

**Comment by t8m** [comment]
@slontis At least for 3.0 (but preferably also all other stable branches) I would prefer if just the test recipe change was backported. Could you please split the existing commit into 2 commits without any other changes so only the test recipe fix could be backported?

**Comment by paulidale** [comment]
Ideally #25262 would go into older branches before this one merges.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**